airbyte-integrations/bases/base-normalization/normalization/transform_catalog/stream_processor.py

```diff
@@ -735,6 +746,36 @@ def generate_scd_type_2_model(self, from_table: str, column_names: Dict[str, Tup
             # SQL Server treats NULL values as the lowest values, then sorted in ascending order, NULLs come first.
             order_null = "desc"
 
+        lag_begin = "lag"
+        lag_end = ""
+        if self.destination_type == DestinationType.CLICKHOUSE:
+            # ClickHouse doesn't support lag() yet, this is a workaround solution
+            # Ref: https://clickhouse.com/docs/en/sql-reference/window-functions/
+            lag_begin = "anyOrNull"
+            lag_end = "ROWS BETWEEN 1 PRECEDING AND 1 PRECEDING"
+
+        case_begin = "case when"
+        case_then = "then"
+        case_else = "else"
+        case_end = "end"
+        if self.destination_type == DestinationType.CLICKHOUSE:
+            # ClickHouse doesn't have CASE WHEN, use multiIf instead
+            # Ref: https://clickhouse.com/docs/en/sql-reference/functions/conditional-functions/#multiif
```

Comment:
but put it in a separate `clickhouse__generate_scd_type_2_model` function will create a large code duplication because most of them are still in common. There are actually only 4 ClickHouse specific parts:

1. `lag()` function
2. `case when`
3. `cast`
4. `left outer join null`

All the other parts are same as other database. Maybe need some code refactoring on `generate_scd_type_2_model` in the future.